src/python/director/drakevisualizer.py

```diff
@@ -215,62 +235,57 @@ def loadPolyDataMeshes(geom):
             print 'warning, cannot find file:', filename
             return []
 
+        meshDataList = []
+
         if filename.endswith('vtm'):
             polyDataList = ioUtils.readMultiBlock(filename)
+            meshDataList = Geometry.DeriveMeshVisList(geom, filename, polyDataList)
         else:
-            polyDataList = [ioUtils.readPolyData(filename)]
-
-        if USE_TEXTURE_MESHES:
-            for polyData in polyDataList:
-                Geometry.loadTextureForMesh(polyData, filename)
-
-
-        Geometry.MeshCache[filename] = list(polyDataList)
-
-        return polyDataList
+            if filename.endswith('obj'):
```

Comment:
i think i need to test this, but i think there is one more remaining bug, which is the case of urdf files that specify visuals with color and mesh obj files, and those obj files do not have associated mtl files.  In this case, the color specified in the urdf should be used.  I think we have urdf files like this.  But as implemented here, the urdf color will be ignored.